canonical-web-and-design/maas-ui
# feat(ui): build LXD project aggregate resources card

## Done

- Built LXD project aggregate resources card, which includes:
  - `ProjectResourcesCard` which is the wrapper card component for all the individual resource sections
  - `VfResources` which displays the "Virtual functions" section
  - `VmResources` which displays the "Total VMs" section
- Note that it doesn't fully match the [design](https://app.zeplin.io/project/601454d5d1141e1b6ea271b7/screen/602aaa0cf4fce564ee6b02c5) for two reasons:
  1. The colours are wrong. It was decided to keep the original colours instead of changing them for this card.
  2. The "Virtual functions" section shows one interface per row instead of two since some interface names can be long. The design of this section is likely to be changed soon since it's possible to have any number of interfaces.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the "Resources" tab of a LXD KVM and check that the project resources card (mostly) matches the [design](https://app.zeplin.io/project/601454d5d1141e1b6ea271b7/screen/602aaa0cf4fce564ee6b02c5). 

## Fixes

Fixes #2228 
Fixes #2434 

## Screenshot

![Screenshot_2021-04-07 LXD resources bolla bolla MAAS](https://user-images.githubusercontent.com/25733845/113808787-1bb4cb80-97aa-11eb-8c06-b5ad0f99b888.png)
